pyispyb/core/routes/webservices/userportalsync.py

```diff
@@ -0,0 +1,48 @@
+import logging
+from fastapi import HTTPException, status, Depends
+from fastapi.responses import Response, JSONResponse
+from pyispyb.app.base import AuthenticatedAPIRouter
+from ...modules import userportalsync as crud
+from ...schemas import userportalsync as schema
+from ....dependencies import permission
+
+
+logger = logging.getLogger("ispyb")
+
+router = AuthenticatedAPIRouter(prefix="/userportalsync", tags=["User Portal Sync"])
+
+
+@router.post(
+    "/sync_proposal",
+    response_model=schema.UserPortalProposalSync,
+)
+def sync_proposal(
+    proposal: schema.UserPortalProposalSync,
+    depends: bool = Depends(permission("uportal_sync")),
+):
+    """Create/Update a proposal from the User Portal and all its related entities"""
+    try:
+        execution_time = crud.sync_proposal(proposal=proposal)
+        proposal_dict = proposal.dict()
+        return JSONResponse(
+            status_code=200,
+            content={
+                "message": f"The proposal {proposal_dict['proposal']['proposalCode']}"
+                f"-{proposal_dict['proposal']['proposalNumber']} has been synchronized in {execution_time}"
+            },
+        )
+    except Exception as e:
```

Comment:
You should also doucment this cases with
```
@router.post(
    ...
    responses={500: {"description": "Something went wrong"}},
```
and this should probably be 400 not 500, as you should know what went wrong, rather than it being an unhandled exception

Reply:
Documented now :)

https://github.com/ispyb/py-ispyb/pull/210/commits/7593bc6be69ed08a33cc0dbb0d28c39a22329890